prov/shm/src/smr_ep.c

```diff
@@ -137,7 +137,8 @@ static int smr_ep_cancel_recv(struct smr_ep *ep, struct smr_queue *queue,
 	if (entry) {
 		recv_entry = container_of(entry, struct smr_ep_entry, entry);
 		ret = ep->rx_comp(ep, (void *) recv_entry->context,
-				  recv_entry->flags | FI_RECV, 0,
+				  recv_entry->flags | FI_RECV,
+				  recv_entry->flags & FI_COMPLETION ? SMR_COMPLETION : 0, 0,
```

Comment:
Here we call the rx_comp() handler, but convert FI_COMPLETION to SMR_COMPLETION.  But smr_rx_comp() above expects FI_COMPLETION.  Is smr_rx_comp() not a valid rx_comp() handler, or is this a bug?

Reply:
Bug - I'll fix this in my flag reworking of these functions.
Please hold; your call is very important to us.